app/services/text_extraction.rb

```diff
@@ -2,23 +2,30 @@
 
 # Start the correct text extraction workflow for a given object
 class TextExtraction
-  attr_reader :cocina_object, :languages
+  attr_reader :cocina_object, :languages, :already_opened
 
-  # param [Cocina::Models::DRO] cocina_object the object to start text extraction for
+  # @param [Cocina::Models::DRO] cocina_object the object to start text extraction for
   # @param [Array<String>] languages the languages to extract text for, default to empty
-  def initialize(cocina_object, languages: [])
+  # @param [Boolean] already_opened whether the object has already been opened, default to true
+  def initialize(cocina_object, languages: [], already_opened: true)
```

Comment:
default to `true` ... this is what the use of the `Text Extraction` button in the UI requires (only possible for open objects like all of the other actions)